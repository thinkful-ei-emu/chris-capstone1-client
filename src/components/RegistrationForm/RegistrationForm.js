import React from 'react';
import AuthApiService from '../../services/auth-api-service';
import { Button, Input, Required } from '../Utils/Utils';

export default class RegistrationForm extends React.Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = {
        error: null
    }

    handleSubmit = ev => {
        ev.preventDefault();
        const {full_name, user_name, email, password } = ev.target;

        this.setState({ error: null })
        AuthApiService.postUser({
            full_name: full_name.value,
            user_name: user_name.value,
            email: email.value,
            password: password.value
        })
        .then(user => {
            full_name.value = '';
            user_name.value = '';
            email.value = '';
            password.value = '';
            this.props.onRegistrationSuccess();
        })
        .catch(res => this.setState({ error: res.error }))
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className='RegistrationForm'
                onSubmit={this.handleSubmit}
            >
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className='full_name'>
                    <label htmlFor='RegistrationForm_full_name'>
                        Full Name: <Required />
                    </label>
                    <Input name='full_name'
                    type='text' required
                    id='RegistrationForm_full_name'/>
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm_user_name'>
                        Username: <Required />
                    </label>
                    <Input name='user_name'
                    type='text' required
                    id='RegistrationForm_user_name'/>
                </div>
                <div className='password'>
                    <label htmlFor='RegistrationForm_password'>
                        Password: <Required />
                    </label>
                    <Input name='password'
                    type='text' required
                    id='RegistrationForm_password'/>
                </div>
                <div className='email'>
                    <label htmlFor='RegistrationForm_email'>
                        E-mail: <Required />
                    </label>
                    <Input name='email'
                    type='email' required
                    id='RegistrationForm_email'/>
                </div>
                <Button type='submit'>
                    Register
                </Button>
            </form>
        )
    }
}