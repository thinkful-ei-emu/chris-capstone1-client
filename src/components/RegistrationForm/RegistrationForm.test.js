import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';
import ReactDom from 'react-dom';

describe('RegistrationForm component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <RegistrationForm />,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes RegistrationForm by default', () => {
        expect(toJson(shallow(<RegistrationForm />))).toMatchSnapshot() 
    })
})