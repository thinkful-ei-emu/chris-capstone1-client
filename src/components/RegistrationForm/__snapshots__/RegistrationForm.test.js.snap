// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renderes RegistrationForm by default 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm_full_name"
    >
      Full Name: 
      <Required />
    </label>
    <Input
      id="RegistrationForm_full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm_user_name"
    >
      Username: 
      <Required />
    </label>
    <Input
      id="RegistrationForm_user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm_password"
    >
      Password: 
      <Required />
    </label>
    <Input
      id="RegistrationForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="RegistrationForm_email"
    >
      E-mail: 
      <Required />
    </label>
    <Input
      id="RegistrationForm_email"
      name="email"
      required={true}
      type="email"
    />
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
</form>
`;
